src/geometry.rs

```diff
@@ -1141,4 +1348,106 @@ mod tests {
 
         assert!(!geometry_with_orphans.has_no_orphan_normals());
     }
+    #[test]
+    fn test_remove_orphan_vertices() {
+        let (faces, vertices, _normals) = quad_with_normals();
+        let extra_vertex = vec![v(0.0, 0.0, 0.0, [0.0, 0.0, 0.0], 1.0)];
+        let vertices_extended = [&extra_vertex[..], &vertices[..]].concat();
+        let faces_renumbered_to_match_extend_vertices: Vec<_> = faces
```

Comment:
Can't you simply append the vertex instead of prepending it so you don't have to renumber? Or is this on purpose?

Reply:
this is on purpose. removing an orphan vertex from the end of the list wouldn't trigger faces to renumber its references into the list of vertices.